content/concepts/did-ddo.md

```diff
@@ -1,173 +1,516 @@
+This document describes how Ocean assets follow the DID/DDO spec, such that Ocean assets can inherit DID/DDO benefits and enhance interoperability.
 
-Decentralized identifiers (DIDs) are a new type of identifier that enables verifiable, decentralized digital identity. Each DID is associated with a unique entity. DIDs may represent humans, objects, and more.
+Decentralized identifiers (DIDs) are a type of identifier that enables verifiable, decentralized digital identity. Each DID is associated with a unique entity. DIDs may represent humans, objects, and more.
 
 A DID Document (DDO) is JSON blob that holds information about the DID. Given a DID, a _resolver_ will return the DDO of that DID.
 
-If a DID is the index key in a key-value pair, then the DID Document is the value to which the index key points.
-The combination of a DID and its associated DID Document forms the root record for a decentralized identifier.
+If a DID is the index key in a key-value pair, then the DID Document is the value to which the index key points. The combination of a DID and its associated DID Document forms the root record for a decentralized identifier.
 
-DIDs and DDOs follow [this specification](https://w3c-ccg.github.io/did-spec/) defined by the World Wide Web Consurtium (W3C).
+DIDs and DDOs follow the [specification defined by the World Wide Web Consortium (W3C)](https://w3c-ccg.github.io/did-spec/).
 
-## Rules for DIDs & DDOs in Ocean
+## Rules for DIDs & DDOs
 
-- An _asset_ in Ocean represents a downloadable file, compute service, or similar. Each asset is a _resource_ under control of a _publisher_. The Ocean network itself does _not_ store the actual resource (e.g. files).
-- An asset should have a DID and DDO. The DDO should include metadata about the asset.
-- The DDO can only can be modified by _owners_ or _delegated users_.
-- There _must_ be at least one client library acting as _resolver_, to get a DDO from a DID.
-- The DDO is stored on-chain. It's stored in in plaintext, with two exceptions: (1) the field for resource-access url is encrypted (2) the whole DDO may be encrypted, if the publisher is willing to lose 100% of discoverability.
-- A metadata cache like Aquarius can help in reading and writing DDO data from the chain.
+An _asset_ in Ocean represents a downloadable file, compute service, or similar. Each asset is a _resource_ under control of a _publisher_. The Ocean network itself does _not_ store the actual resource (e.g. files).
 
-## DID Structure
+An _asset_ should have a DID and DDO. The DDO should include [metadata](#metadata) about the asset, and define access in at least one [service](#services). The DDO can only can be modified by _owners_ or _delegated users_.
 
-In Ocean, a DID is a string that looks like:
+There _must_ be at least one client library acting as _resolver_, to get a DDO from a DID. A metadata cache like Aquarius can help in reading and searching through DDO data from the chain.
```

Comment:
I do not understand what's meant with this first sentence. This almost sounds like I can only resolve with e.g. ocean.js and should not query Aquarius. I'm sure we can find a more clear formulation here

Reply:
I believe the meaning is that you can't read directly from the chain, thus needing a "client library". You can skip these steps by using `Aquarius` . I agree this is confusing.

Reply:
ok, should be more clear now, just removed sentence and next section then talks about all the encryption.